var/spack/repos/builtin/packages/fmt/package.py

```diff
@@ -79,6 +79,8 @@ class Fmt(CMakePackage):
     # (https://github.com/fmtlib/fmt/issues/3028)
     conflicts("cxxstd=17", when="@9.0.0%intel")
     conflicts("cxxstd=17", when="@9.0.0%nvhpc")
+    # clang-21 requires fmt-11.1.0 (https://github.com/fmtlib/fmt/pull/4187)
+    conflicts("%llvm@21:", when="@:11.0")
```

Comment:
Only thing I'm not sure about is if this should be
```suggestion
    conflicts("%[virtuals=cxx] llvm@21:", when="@:11.0")
```
nowadays with compilers as nodes (I'm not sure why the `depends_on("c")` is there, but that's orthogonal...).

@alalazo @haampie does it make a difference?

Reply:
> does it make a difference?

Yes. `%llvm` means "llvm is a direct dependency", `%[virtuals=cxx] llvm` means "llvm is a direct dependency AND provides cxx"